app/controllers/application_controller.rb

```diff
@@ -308,9 +309,15 @@ def current_email
     current_user&.email
   end
 
-  def set_locale
-    if ENV.fetch('LOCALIZATION_ENABLED', 'false') == 'true'
-      I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
+  def switch_locale(&action)
+    locale = nil
+    if cookies[:locale]
+      locale = cookies[:locale]
+    elsif ENV.fetch('LOCALIZATION_ENABLED', 'false') == 'true'
```

Comment:
Généralement quand on a une variable d'env `*_ENABLED` on fait `ENV['LOCALIZATION_ENABLED'] == 'enabled'`

Reply:
@LeSim est ce qu'il y avait une raison particulière pour que tu l'es écrit de cette manière (je fais référence au commit "[use env for localization to avoid db round trip]([https://github.com/betagouv/demarches-simplifiees.fr/commit/b87a691176e4e18a87332e30a108e01935b8e76a#diff-766c34fd6533171eaf54300c153f89d6002c35c02cfc9c5b219251f85180ad07])")?

Reply:
non, c'est une erreur de ma part. Le but a l'époque était juste de limiter les accès db. +1 pour la reco de tchak mais si ça veut potentiellement dire changer les variables d'env chez nous avant le déploiement